programs/marinade-finance/src/instructions/crank/update.rs

```diff
@@ -241,6 +241,26 @@ impl<'info> UpdateCommon<'info> {
         }
         Ok(())
     }
+
+    #[inline]
+    pub fn update_msol_price(&mut self) -> Result<()> {
+        // price is computed as:
+        // total_active_balance + total_cooling_down + reserve - circulating_ticket_balance
+        // DIVIDED by msol_supply
+        self.state.msol_price = self
+            .state
+            .calc_lamports_from_msol_amount(State::PRICE_DENOMINATOR)?; // store binary-denominated mSOL price
+        Ok(())
+    }
+
+    pub fn mint_protocol_fees(&mut self, lamports_incoming: u64) -> Result<()> {
```

Comment:
Return protocol_rewards_fee and fee_as_msol_amount values because I need to include them in the event logs. And don't use msg! because I will use an event instead

Reply:
vityc needs the msg!s, event will be additional